src/test/training/supportbank/BankTest.java

```diff
@@ -0,0 +1,65 @@
+package training.supportbank;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import training.supportbank.Account;
+import training.supportbank.Bank;
+import training.supportbank.Transaction;
+
+import java.math.BigDecimal;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class BankTest {
+
+    Bank bank;
+
+    @BeforeEach
+    public void createBank() {
+        bank = new Bank();
+    }
+
+    @Test
+    public void addMulitpleAccountsDoesNotAddAccountIfNameAlreadyExists() {
+        Set<String> accounts = Sets.newHashSet("accountName", "anotherAccount");
+        bank.addMultipleAccounts(accounts);
+        Transaction transaction = new Transaction("01/01/2014", "accountName", "anotherAccount", "narrative", "10");
+        List<Transaction> transactions = new ArrayList<>();
+        transactions.add(transaction);
+        bank.processMultipleTransactions(transactions);
+        Set<String> newAccounts = Sets.newHashSet("accountName");
+        bank.addMultipleAccounts(newAccounts);

```

Comment:
This test is quite long, but I suspect that if you inlined a lot of these variables it would be easier to follow, e.g.
```
bank.processMultipleTransactions(Lists.newArrayList(transaction));
```